semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/http/HttpSemanticConventionUtils.java

```diff
@@ -433,6 +450,17 @@ public static URL getNormalizedUrl(String url) throws MalformedURLException {
     return Optional.empty();
   }
 
+  private static String getProtocolFromForwarded(String value) {
+    String[] values = value.split(";");
+    for (String subValue : values) {
+      String subValueWithoutSpaces = subValue.replaceAll("\\s", "");
```

Comment:
Looking at the test example `by=random;proto=http`, would simply trim work? 
Would something like below?
```
private static String getProtocolFromForwarded(String value) {
    String[] values = value.split(";");
    for (String subValue : values) {
      String[] protoSplits = subValue.trim().split("=");
      if (protoSplits.length == 2 && protoSplits[0].equals("proto")) {
        return protoSplits[1];
      }
    }
    return "";
  }
 ```
Or if we need to trim further too?
```
private static String getProtocolFromForwarded(String value) {
    String[] values = value.split(";");
    for (String subValue : values) {
      String[] protoSplits = subValue.trim().split("=");
      if (protoSplits.length == 2 && protoSplits[0].trim().equals("proto")) {
        return protoSplits[1].trim();
      }
    }
    return "";
  }
```

Reply:
We may have something like `by =random ; proto = http` so the second approach will work.

Reply:
Refactored the code. Using the second approach.